.github/workflows/test-e2e.yml

```diff
@@ -144,11 +144,20 @@ jobs:
           done
       - name: Checkout template-operator
         uses: actions/checkout@v3
-        if: ${{ matrix.e2e-test == 'module-status-propagation' }}
+        if: ${{ matrix.e2e-test == 'module-status-propagation' || matrix.e2e-test == 'non-blocking-deletion' }}
         with:
           repository: kyma-project/template-operator
           path: ./template-operator/
       - name: Create Template Operator Module
+        working-directory: ./template-operator
+        if: ${{ matrix.e2e-test == 'non-blocking-deletion' }}
+        run: |
+          kyma alpha create module --module-config-file ./module-config-no-cr.yaml --path . --registry k3d-kcp-registry:5111 --insecure --name kyma-project.io/module/template-operator
```

Comment:
I wonder if we should create a subdirectory `/tests/e2e_test/scripts` in our project, create files with shell scripts there and then reference those scripts in here. For example:

```yml
- name: Create Template Operator Module
        working-directory: ./template-operator
        if: ${{ matrix.e2e-test == 'non-blocking-deletion' }}
        run: |
            $PATH_TO_SCRIPTS/deploy_module.sh
```

@ruanxin what do you think?

Reply:
make sense, and especially if the script can be reused, for example, here this `deploy_module.sh` should be possible to accept some params, so that it can be reused in followup "Create Template Operator Module with final state `Warning`" step.

We can either progress this refactoring here or create a followup issue to address this, it is up to you @jeremyharisch .

Reply:
I like the idea, I will create an issue for that - I do not want to blow  this PR up even up

Reply:
Follow-up issue https://github.com/kyma-project/lifecycle-manager/issues/945